src/AzureAIProxy/Services/AssistantService.cs

```diff
@@ -0,0 +1,96 @@
+using AzureAIProxy.Shared.Database;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Caching.Memory;
+using System.Text.Json;
+
+namespace AzureAIProxy.Services;
+
+/// <summary>
+/// Provides services for managing assistant assets, including adding, deleting, and retrieving assistant records.
+/// <param name="db">The database context for accessing assistant records.</param>
+/// <param name="memoryCache">The memory cache for caching assistant records.</param>
+/// </summary>
+public class AssistantService(AzureAIProxyDbContext db, IMemoryCache memoryCache) : IAssistantService
+{
+
+    /// <summary>
+    /// Adds a new assistant record to the database based on the provided API key and response content.
+    /// </summary>
+    /// <param name="api_key">The API key associated with the assistant.</param>
+    /// <param name="responseContent">The response content containing the assistant ID.</param>
+    /// <returns>A task that represents the asynchronous operation.</returns>
+    public async Task AddIdAsync(string api_key, string responseContent)
+    {
+        var jsonElement = JsonDocument.Parse(responseContent).RootElement;
+        var id = jsonElement.TryGetProperty("id", out var idElement) ? idElement.GetString() : null;
+        if (id is null) return;
+
+        var assistant = new Assistant
+        {
+            ApiKey = api_key,
+            Id = id
+        };
+
+        db.Assistants.Add(assistant);
+        await db.SaveChangesAsync();
+    }
+
+    /// <summary>
+    /// Deletes an assistant record from the database based on the provided API key and response content.
+    /// </summary>
+    /// <param name="api_key">The API key associated with the assistant.</param>
+    /// <param name="responseContent">The response content containing the assistant ID and deletion status.</param>
+    /// <returns>A task that represents the asynchronous operation.</returns>
+    public async Task DeleteIdAsync(string api_key, string responseContent)
+    {
+        var jsonElement = JsonDocument.Parse(responseContent).RootElement;
```

Comment:
Like above, can't we pass in a type from the routing engine rather than parsing JSON ourselves?